File: C++/isPrime.cpp

#include <iostream>

using namespace std;

int main(){
  
  int isPrime;
  int guess=2;
  
  cout << "Enter number to test if it is prime: ";
  cin >> isPrime;
  
  while(isPrime >= guess){
    
    if( guess != isPrime && isPrime%guess == 0 ){
      
      cout << isPrime << " is not a prime number!" << endl;
      break;
      
    }else if(guess==isPrime){
      
      cout << isPrime << " is a prime number!"<< endl;
      break;
      
    }else{
      
      guess++;

    }
  }
  
return 0;
  
}

File: Go/isPrime.go
/*
Purpose: Test if an inputted number is prime
*/

package main

import "fmt"

func isPrime(input int) bool {
	for divisor := 2; divisor <= input/2; divisor++ {
		if input%divisor == 0 {
			return false
		}
	}
	return true
}

func main() {
	var input int

	fmt.Print("Enter number: ")
	fmt.Scanln(&input)

	isPrime := isPrime(input)

	fmt.Printf("%d is Prime: %v\n", input, isPrime)
}
